core/vtk/ttkMergeTreeDistanceMatrix/ttkMergeTreeDistanceMatrix.cpp

```diff
@@ -100,41 +100,85 @@ int ttkMergeTreeDistanceMatrix::run(
   std::vector<vtkSmartPointer<vtkMultiBlockDataSet>> &inputTrees,
   std::vector<vtkSmartPointer<vtkMultiBlockDataSet>> &inputTrees2) {
 
-  // Construct trees
-  const int numInputs = inputTrees.size();
-  std::vector<MergeTree<dataType>> intermediateTrees, intermediateTrees2;
-  constructTrees(inputTrees, intermediateTrees);
-  constructTrees(inputTrees2, intermediateTrees2);
-
   // Verify parameters
   if(not UseFieldDataParameters) {
     if(Backend == 0) {
       branchDecomposition_ = true;
       normalizedWasserstein_ = true;
       keepSubtree_ = false;
+      baseModule = 0;
     } else if(Backend == 1) {
       branchDecomposition_ = false;
       normalizedWasserstein_ = false;
       keepSubtree_ = true;
+      baseModule = 0;
+    } else if(Backend == 3) {
+      branchDecomposition_ = true;
+      normalizedWasserstein_ = false;
+      keepSubtree_ = true;
+      baseModule = 1;
+    } else if(Backend == 4) {
+      branchDecomposition_ = true;
+      normalizedWasserstein_ = false;
+      keepSubtree_ = true;
+      baseModule = 2;
+    } else {
+      baseModule = 0;
     }
   }
-  if(not branchDecomposition_) {
-    if(normalizedWasserstein_)
-      printMsg("NormalizedWasserstein is set to false since branch "
-               "decomposition is not asked.");
-    normalizedWasserstein_ = false;
+  if(baseModule == 0) {
```

Comment:
Would it be possible to have this logic implemented in the base code instead?
(by logic, I mean the check for compatibility between computation options).

Otherwise, the base code becomes very difficult to use (independently of VTK).

(maybe this is not easy to do because there are so many unrelated base code objets)

Reply:
This is a good point, I think that would make sense. However, the code you are referring to here is actually Mathieu's. The only integrity check I added is in the MergeTreeClustering module (here I only added console outputs). Maybe I should discuss this with Mathieu first and change this later?
@julien-tierny @MatPont

Reply:
@julien-tierny I just talked to Mathieu and he agrees that it makes sense to move these checks to the base layer. But we have to move some member variables as well and do some restructuring, so we think it makes more sense to do this in another PR.

Reply:
sounds good for another PR then.
that'd be great not to forget about it in the long run though :D